NamelessSpigot/src/com/namelessmc/plugin/spigot/WhitelistRegistered.java

```diff
@@ -38,15 +40,21 @@ public void run() {
 				e.printStackTrace();
 				return;
 			}
+			
+			Config.MAIN.getConfig().getStringList("permanent-whitelisted")
+			.forEach(playername -> permWhitelistUUID.add(playername.getUniqueId()));		
 
 			Bukkit.getScheduler().runTask(NamelessPlugin.getInstance(), () -> {
 				for (final OfflinePlayer whitelistedPlayer : Bukkit.getWhitelistedPlayers()) {
 					if (!uuids.contains(whitelistedPlayer.getUniqueId())) {
-						// The player is whitelisted, but no(t) (longer) registered.
-						whitelistedPlayer.setWhitelisted(false);
-						uuids.remove(whitelistedPlayer.getUniqueId());
-						if (log) {
-							logger.info("Removed " + whitelistedPlayer.getName() + " from the whitelist.");
+						// The player is not on the permanent whitelist exclusion list
+						if (!permanentWhitelisted.contains(whitelistedPlayer.getUniqueId())) {						
```

Comment:
To avoid indenting all code below, I'd use this instead:
```java
if (exclude.contains(whitelistedPlayer.getUniqueId()) {
    continue;
}
```
and put it above the other if statement so it's the first if statement in the for loop